subsys/bluetooth/audio/mcc.c

```diff
@@ -890,19 +923,35 @@ static uint8_t mcs_notify_handler(struct bt_conn *conn,
 				  const void *data, uint16_t length)
 {
 	uint16_t handle = params->value_handle;
+	struct mcs_instance_t *mcs_inst;
+
+	if (data == NULL) {
+		BT_DBG("[UNSUBSCRIBED] 0x%04X", params->value_handle);
+		params->value_handle = 0U;
 
-	if (conn == NULL) {
+		return BT_GATT_ITER_CONTINUE;
+	}
+
+	mcs_inst = lookup_inst_by_conn(conn);
+	if (mcs_inst == NULL) {
 		return BT_GATT_ITER_CONTINUE;
 	}
 
 	BT_DBG("Notification, handle: %d", handle);
 
-	if (data) {
-		if (handle == cur_mcs_inst->player_name_handle) {
+	if (mcs_inst != NULL) {
```

Comment:
You check if mcs_inst == NULL on line 936, so no need to check again since I assume this is mostly run in BT_RX thread and can't be interrupted